Improvements to EditStream and Input

agzuniverse opened the pull request:
Use styles from utilityStyles instead of styles defined locally for margins in EditStream, and set `underlineColorAndroid` of input to `BORDER_COLOR` on focus and `HALF_COLOR` when not focused. Previously the color was black, which didn't look right in night mode. Does not affect search input, composeBox input etc as the bottom border is not visible in these.

After changes the screen looks like this:
![screenshot_20180531-210555](https://user-images.githubusercontent.com/22353313/40792349-5d264c06-6517-11e8-8213-16871d1a84a2.png)

borisyankov commented on src/streams/EditStreamCard.js:
```diff
@@ -68,20 +63,21 @@ export default class EditStreamCard extends PureComponent<Props, State> {
         <Label text="Name" />
         <Input
-          style={styles.marginBottom}
+          style={styles.halfMarginBottom}
```
Even though `halfMarginBottom` which is `8` is closer to the older value of `10`, you should use the new utility one of `marginBottom` that is `16`.

agzuniverse replied:
using `marginBottom` causes the view to become scrollable (it doesn't seem to expand in height) and will require further changes to correct that.

Although that is probably be a good thing since that won't become an issue if anything is added to this screen later that causes it to exceed the current view height.
I will the changes required and update the PR.

agzuniverse commented:
Changed the margin from `halfMarginBottom` to `marginBottom` (8 to 16)

Also modified `Input.js` to have an `onBlur` and `onFocus` methods to properly control the input underline color. This does not interfere with the onBlur and onFocus functions passed as a prop by other components that use `Input`.

gnprice commented:
Thanks @agzuniverse ! Just pushed the first of these commits.

To finish reviewing the other I want to play a bit with things on my iPhone, which is out of battery; just started it charging, and I'll come back to this.

gnprice commented:
And merged the second commit too! I just expanded the commit message a bit (based partly on your PR message) to explain why this doesn't affect the compose and search boxes, because that puzzled me for a bit. Thanks again for the fix.
